strax/processing/peak_building.py

```diff
@@ -255,6 +278,31 @@ def sum_waveform(peaks, records, adc_to_pe, select_peaks_indices=None):
         p['area_per_channel'][:] = area_per_channel
 
 
+@numba.njit(cache=True, nogil=True)
+def _build_hit_waveform(hit, record, hit_waveform):
+    """
+    Adds information for overlapping record and hit to hit_waveform.
+    Updates hit_waveform inplace. Result is still in ADC counts.
+
+    :returns: Boolean if record saturated within the hit.
+    """
+    (h_start_record, h_end_record), (r_start, r_end) = strax.overlap_indices(
+        hit['time'] // hit['dt'], hit['length'],
+        record['time'] // record['dt'], record['length'])
+
+    # Get record properties:
+    record_data = record['data'][r_start:r_end]
+    multiplier = 2**record['amplitude_bit_shift']
+    bl_fpart = record['baseline'] % 1
```

Comment:
I was wondering if we could re-use https://github.com/AxFoundation/strax/blob/e1b77aa7dd9b951dc14f0066f3e4e900ff4833ab/strax/processing/pulse_processing.py#L110. But it seems not. However, it did make me realize that according to this definition, the 'area' of a record is not really going to make much sense as we are also adding the floating part of the BL there for the entire length of the record.

Not very much a comment to your PR, just a realization.